pkg/pxc/app/statefulset/haproxy.go

```diff
@@ -217,6 +233,23 @@ func (c *HAProxy) SidecarContainers(spec *api.PodSpec, secrets string, cr *api.P
 			MountPath: "/etc/mysql/mysql-users-secret",
 		})
 	}
+	if cr.CompareVersionWith("1.9.0") >= 0 {
+		fvar := true
+		container.EnvFrom = []corev1.EnvFromSource{
+			{
+				SecretRef: &corev1.SecretEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cr.Spec.HAProxy.EnvVarsSecretName,
+					},
+					Optional: &fvar,
+				},
+			},
+		}
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
```

Comment:
should we do VolumeMounts for PXC and ProxySQL?

Reply:
For HAProxy we have use case for that moment that is why I have added it. For PXC and ProxySQL we do not have it for now. So, we can add it right now or wait until we have it.  @delgod  as you wish.

Reply:
up to you